x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -298,15 +359,75 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
           dataViewId === null
             ? []
             : getFiltersForDSLQuery(job.datafeed_config.query, dataViewId, job.job_id),
-      });
+        ...(withWindowParameters
+          ? {
+              wp: { bMin, bMax, dMin, dMax },
+            }
+          : {}),
+      };
+    };
+
+    const generateDiscoverUrl = async () => {
+      const pageState = await generateRedirectUrlPageState();
+      const url = await discoverLocator.getRedirectUrl(pageState);
 
       if (!unmounted) {
         setOpenInDiscoverUrl(url);
       }
     };
 
+    const generateLogRateAnalysisUrl = async () => {
+      if (
+        !LOG_RATE_ANALYSIS_ML_FUNCTIONS.includes(
+          props.anomaly.source.function as LogRateAnalysisMlFunctions
+        ) &&
+        !unmounted
+      ) {
+        setOpenInLogRateAnalysisUrl(undefined);
+        return;
+      }
+
+      const mlLocator = share.url.locators.get(ML_APP_LOCATOR);
+
+      if (!mlLocator) {
+        // eslint-disable-next-line no-console
+        console.error('Unable to detect locator for ML or bounds');
+        return;
+      }
+      const pageState = await generateRedirectUrlPageState(true, 'time');
+
+      const { indexPatternId, wp, query, ...globalState } = pageState;
```

Comment:
For accurate results in the log rate analysis view, you should also get the datafeed query out of the job config and pass that to the target page.

@jgowdyelastic are we doing this for the anomalies table action for log pattern analysis.

Reply:
The log pattern analysis flyout does not use the datafeed's query. 
It does use the partition field to filter the pattern analysis, but in sub-agg.  We get all of the patterns in the full time range and then use a sub agg to reduce the categories to the ones in the bucket and filtered by partition field name and value.

The problem with pattern analysis is we really need to run over the most data we can to get accurate categories, and then filter down.
By adding the datafeed query to the pattern analysis query we'd be reducing the data seen and so we may get inaccurate categories.

Reply:
This is now fixed in the latest update: The links to Discover and Log Rate Analysis are generated using the same code so we make use of the same helper (`getFiltersForDSLQuery()`) to apply the datafeed query to the generated link. Here's an example how the filter ends up being applied in the search bar in Log Rate Analysis:

<img width="1348" alt="image" src="https://github.com/elastic/kibana/assets/230104/9fe84107-4a1c-40b1-b8a6-0255d61518dc">